forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabForgeFactory.java

```diff
@@ -43,6 +47,33 @@ public String name() {
         return Set.of("gitlab.com");
     }
 
+    private void configureSshKey(String userName, String hostName, String sshKey) {
+        var cfgPath = Path.of(System.getProperty("user.home"), ".ssh");
+        var cfgFile = cfgPath.resolve("config");
+        var existing = "";
+        try {
+            existing = Files.readString(cfgFile, StandardCharsets.UTF_8);
+        } catch (IOException ignored) {
+        }
+
+        var userHost = userName + "." + hostName;
+        var existingBlock = Pattern.compile("^Match host=" + Pattern.quote(userHost) + "(?:\\R[ \\t]+.*)+", Pattern.MULTILINE);
+        var existingMatcher = existingBlock.matcher(existing);
+        var filtered = existingMatcher.replaceAll("");
+        var result = "Match host=" + userHost + "\n" +
```

Comment:
I don't think this is valid .ssh/config syntax. Afaik it should be `Match <keyword> <value>`, that is, no `=`. I see no tests in this PR. Are you sure this works?

Reply:
Yes, tested it manually and seems to work fine. And no, there are no automated tests that exercise GitLab or GitHub APIs, except for running them with a special credential configuration..

Reply:
I still think this does not follow the documented ssh config file syntax (https://man7.org/linux/man-pages/man5/ssh_config.5.html). And even if that works on your machine right now, it can break on a future update of ssh.

Reply:
> Configuration options may be separated by whitespace or optional whitespace and exactly one ‘=’

Still, works without it as well, so can certainly change it..